core/src/main/java/hudson/EnvVars.java

```diff
@@ -74,6 +74,7 @@
  * @author Kohsuke Kawaguchi
  */
 public class EnvVars extends TreeMap<String,String> {
+    private static final long serialVersionUID = 4320331661987259022L;
```

Comment:
Note that despite I fully agree this is a good idea to use the value that's generated by default in the errors you've seen,  FWIW IIRC the generated value is not defined in the spec, i.e. it could still fail on various JDKs implementations.

But anyway, we cannot do better than that, and at least this will make it fixed for all implementations in the future so :+1:.

Also, if we go this path, should we then review all Serializable classes in Jenkins Core? Before this issue arises on other classes?

Looking https://github.com/jenkinsci/jenkins/search?q=Serializable&unscoped_q=Serializable, it seems we would ideally need to patch a lot of others isn't it?

Reply:
"_the generated value is not defined in the spec_"
Interesting, I wasn't aware of that.

"_should we then review all Serializable classes_"
I guess yes, ideally. Though its probably even more than that list as in this case the class only implements `Serializable` indirectly.